job_definitions/release_application_paas.yml

```diff
@@ -53,14 +53,33 @@
               }
 
               stage('Deploy') {
-                withEnv(["DM_CREDENTIALS_REPO=/var/lib/jenkins/digitalmarketplace-credentials", "CF_HOME=${pwd()}"]) {
-                  paas_credentials = sh(script: '$DM_CREDENTIALS_REPO/sops-wrapper -d $DM_CREDENTIALS_REPO/jenkins-vars/paas_credentials_env.enc', returnStdout: true).trim()
-                  withEnv(paas_credentials.tokenize("\n")) {
-                      sh('make paas-login')
+                  withEnv(["DM_CREDENTIALS_REPO=/var/lib/jenkins/digitalmarketplace-credentials", "CF_HOME=${pwd()}"]) {
+                    paas_credentials = sh(script: '$DM_CREDENTIALS_REPO/sops-wrapper -d $DM_CREDENTIALS_REPO/jenkins-vars/paas_credentials_env.enc', returnStdout: true).trim()
+                    withEnv(paas_credentials.tokenize("\n")) {
+                        sh('make paas-login')
+                    }
                   }
-                  lock("paas-${STAGE}-${APPLICATION_NAME}-deploy") {
-                      sh('make ${STAGE} deploy-app')
+
+                lock("paas-${STAGE}-${APPLICATION_NAME}-deploy") {
+                  if ("$RELEASE_NAME" == "") {
+                    echo "No release specified. Detecting currently deployed release version of ${APPLICATION_NAME} ..."
+
+                    /*
+                      This is a pretty painful command to parse. It essentially does the following:
+
+                      1) Hits `cf app ${APPLICATION_NAME} --guid` to get the CloudFoundry GUID of an application.
+                      2) Uses `cf curl /v2/apps/guid/routes` to work out the subdomain/path suffix an application is registered on.
```

Comment:
Should this be subdomain or host? It's weird, my understanding is that `cloudapps.digital` is the host and `dm-api-preview`, for example, is the subdomain. But the cf api seems to use 'host' for what I thought was the subdomain.

Reply:
Yeah the terminology gets pretty mixed up between reality/CF... Will change later references to subdomain.